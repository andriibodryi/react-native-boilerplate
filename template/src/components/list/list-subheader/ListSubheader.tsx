import { FC } from 'react';

import Text from '@src/components/typography/text/Text';

import { styles } from './list-subheader.presets';
import { ListSubheaderTypes } from './list-subheader.types';

const ListSubheader: FC<ListSubheaderTypes> = ({ style, ...rest }) => (
  <Text
    preset="textMdMedium"
    numberOfLines={1}
    {...rest}
    style={[styles.base, style]}
  />
);

export default ListSubheader;
