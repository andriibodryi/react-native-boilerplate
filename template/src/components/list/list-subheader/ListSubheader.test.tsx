import React from 'react';
import renderer from 'react-test-renderer';

import ListSubheader from './ListSubheader';

it('should render list subheader correctly', () => {
  const tree = renderer.create(<ListSubheader />).toJSON();

  expect(tree).toMatchSnapshot();
});
