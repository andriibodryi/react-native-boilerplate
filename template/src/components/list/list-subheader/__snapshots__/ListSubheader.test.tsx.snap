// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render list subheader correctly 1`] = `
<Text
  numberOfLines={1}
  style={
    Array [
      Object {
        "color": "#4D4D4D",
        "fontFamily": "Aspira-Regular",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 20,
      },
      Array [
        Object {
          "color": "#4D4D4D",
          "fontFamily": "Aspira-Medium",
          "fontSize": 14,
          "fontWeight": "500",
        },
        Object {},
        Object {
          "color": "#4D4D4D",
        },
        Array [
          Object {
            "color": "#4D4D4D",
            "fontSize": 12,
            "letterSpacing": 1.5,
            "paddingBottom": 8,
          },
          undefined,
        ],
      ],
    ]
  }
/>
`;
